nbconvert/preprocessors/execute.py

```diff
@@ -36,7 +36,13 @@ class ExecutePreprocessor(Preprocessor):
     """
 
     timeout = Integer(30, config=True,
-        help="The time to wait (in seconds) for output from executions."
+        help=dedent(
+            """
+            The time to wait (in seconds) for output from executions.
+            If a cell execution takes longer, a `CellExecutionError`
```

Comment:
Looking at the code, it actually raises `TimeoutError` on Python 3 and `RuntimeError` on Python 2 (because TimeoutError is new in Python 3)